Packs/Syslog/Integrations/Syslogv2/Syslogv2.py

```diff
@@ -0,0 +1,335 @@
+    while True:
+        try:
+            line = file_obj.readline()
+            if not line:
+                demisto.info(f'Disconnected from {address}')
+                break
+            perform_long_running_loop(line.strip())
+        except Exception as e:
+            demisto.error(traceback.format_exc())  # print the traceback
+            demisto.error(f'Error occurred during long running loop. Error was: {e}')
+        finally:
+            demisto.debug('Finished reading message')
+    file_obj.close()
+
+
+def prepare_globals_and_create_server(port: int, log_format: str, message_regex: Optional[str],
+                                      incident_type: Optional[str], certificate: Optional[str],
+                                      private_key: Optional[str]) -> StreamServer:
+    """
+    Prepares global environments of LOG_FORMAT, MESSAGE_REGEX, INCIDENT_TYPE and creates the server to listen
+    to Syslog messages.
+    Args:
+        port (int): Port
+        log_format (str): Log format of messages to be received from Syslog, either RFC-3164 or RFC-5424.
+        message_regex (Optional[str]): Regex. Will create incident only if Syslog message matches this regex.
+        incident_type (Optional[str]): Incident type.
+        certificate (Optional[str]): Certificate. For SSL connection.
+        private_key (Optional[str]): Private key. For SSL connection.
+
+    Returns:
+        (StreamServer): Server to listen to Syslog messages.
+    """
+    global LOG_FORMAT, MESSAGE_REGEX, INCIDENT_TYPE
+    LOG_FORMAT = log_format
+    MESSAGE_REGEX = message_regex
+    INCIDENT_TYPE = incident_type
+    if certificate and private_key:
+        certificate_file = NamedTemporaryFile(delete=False)
+        certificate_path = certificate_file.name
+        certificate_file.write(bytes(certificate, 'utf-8'))
+        certificate_file.close()
+
+        private_key_file = NamedTemporaryFile(delete=False)
+        private_key_path = private_key_file.name
+        private_key_file.write(bytes(private_key, 'utf-8'))
+        private_key_file.close()
+        server = StreamServer(('0.0.0.0', port), perform_long_running_execution, keyfile=private_key_path,
+                              certfile=certificate_path)
+        demisto.debug('Starting HTTPS Server')
+    else:
+        server = StreamServer(('0.0.0.0', port), perform_long_running_execution)
+        demisto.debug('Starting HTTP Server')
+    return server
+
+
+def get_mapping_fields() -> Dict[str, str]:
+    def transfer_type_to_str_type(type_: Any):
+        type_as_str: str = str(type_)
+        # remove typing from the type name
+        type_as_str = type_as_str.replace('typing.', '')
+        # For classes other than typing, eg <class 'str'>
+        type_as_str = type_as_str.replace('<class ', '').replace("'", '').replace('>', '')
+        return type_as_str
+
+    return {k: transfer_type_to_str_type(v) for k, v in get_type_hints(SyslogMessageExtract).items()}
```

Comment:
nice impl, but better to provide the field description in the values of the dict, or which is used in which format. you can do it hard-coded.